scripts/irods/test/test_iput.py

```diff
@@ -541,6 +541,79 @@ def test_iput_with_apostrophe_logical_path__issue_5759(self):
 
             self.user.assert_icommand(['ils', '-l', collection_path], 'STDOUT', coll)
 
+    def test_recursive_bulk_overwrite_is_not_allowed__issue_7110(self):
+        filenames = ['file1', 'file2']
+        bonus_filename = 'file3'
+
+        dirname = 'bulkputme'
+        directory_path = os.path.join(self.user.local_session_dir, dirname)
+
+        collection_path = os.path.join(self.user.session_collection, dirname)
+        logical_paths = [os.path.join(collection_path, f) for f in filenames]
+        bonus_logical_path = os.path.join(collection_path, bonus_filename)
+
+        random_resource = 'rand'
+        ufs_resource_1 = 'ufs1'
+        ufs_resource_2 = 'ufs2'
+
+        try:
+            # Create the two files so that we can perform the bulk put.
+            os.mkdir(directory_path)
+            for f in filenames:
+                physical_path = os.path.join(directory_path, f)
+                lib.make_arbitrary_file(physical_path, 2)
+
+            # Set up random resource hierarchy such that only 1 replica will appear.
+            lib.create_random_resource(self.admin, random_resource)
+            lib.create_ufs_resource(self.admin, ufs_resource_1, hostname=test.settings.HOSTNAME_2)
+            lib.create_ufs_resource(self.admin, ufs_resource_2, hostname=test.settings.HOSTNAME_3)
+            lib.add_child_resource(self.admin, random_resource, ufs_resource_1)
+            lib.add_child_resource(self.admin, random_resource, ufs_resource_2)
+
+            # Bulk put the directory ensure that the objects appear as expected.
+            self.user.assert_icommand(
+                ['iput', '-R', random_resource, '-br', directory_path, self.user.session_collection], 'STDOUT')
+
+            for lp in logical_paths:
+                self.assertFalse(lib.replica_exists(self.user, lp, 1))
+                self.assertEqual(str(1), lib.get_replica_status(self.user, os.path.basename(lp), 0))
+
+            # Add a bonus file to the directory so that the overwrite has something to do.
+            lib.make_arbitrary_file(os.path.join(directory_path, bonus_filename), 2)
+
+            # Bulk put the directory with force flag to overwrite and ensure that the action is disallowed.
+            self.user.assert_icommand(
+                ['iput', '-R', random_resource, '-brf', directory_path, self.user.session_collection],
+                'STDERR', 'USER_INCOMPATIBLE_PARAMS')
```

Comment:
Consider including the error code with the error name to avoid false test results.

Reply:
Will do

Reply:
Done